# Supply Chain Analytics in Tableau or Power BI

## 📖 Background
Test your BI skills on a real-world dataset focusing on supply chain analytics. As the main data analyst for Just In Time, you will help solve key shipment and inventory management challenges, analyze supply chain inefficiencies, and create insightful dashboards to inform business stakeholders about potential problems and propose structural business improvements.

Be creative and make use of your full skillset! Use this Workspace to prepare your data, import the tables into your local Tableau or Power BI instance, and share your insights below.

The end goal will be a (set of) interactive dashboards that demonstrate clear insights for Just In Time.

## 💾 The data


| Group | Column name | Dataset | Definition |
|:---|:---|:---|:---|
| Customer | Customer ID | orders_and_shipments.csv | Unique customer identification |
| Customer | Customer Market | orders_and_shipments.csv | Geographic grouping of customer countries, with values such as Europe, LATAM, Pacific Asia, etc. |
| Customer | Customer Region | orders_and_shipments.csv | Geographic grouping of customer countries, with values such as Northern Europe, Western Europe, etc. |
| Customer | Customer Country | orders_and_shipments.csv | Customer's country |
| Order info | Order ID | orders_and_shipments.csv | Unique Order identification. Order groups one or multiple Order Items |
| Order info | Order Item ID | orders_and_shipments.csv | Unique Order Item identification. Order Item always belong to just one Order |
| Order info | Order Year | orders_and_shipments.csv | Year of the order |
| Order information | Order Month | orders_and_shipments.csv | Month of the order |
| Order information | Order Day | orders_and_shipments.csv | Day of the order |
| Order information | Order Time | orders_and_shipments.csv | Timestamp of the order in UTC |
| Order information | Order Quantity | orders_and_shipments.csv | The amount of items that were ordered within a given Order Item (1 record of the data) |
| Product | Product Department | orders_and_shipments.csv | Product grouping into categories such as Fitness, Golf, Pet Shop, etc. |
| Product | Product Category | orders_and_shipments.csv | Product grouping into categories such as Sporting Goods, Women's Apparel, etc. |
| Product | Product Name | orders_and_shipments.csv | The name of the purchased product |
| Sales | Gross Sales | orders_and_shipments.csv | Revenue before discounts generated by the sales of the Order Item (1 record of the data) |
| Sales | Discount % | orders_and_shipments.csv | Discount % applied on the catalog price |
| Sales | Profit | orders_and_shipments.csv | Profit generated by the sales of the Order Item (1 record of data) |
| Shipment information | Shipment Year | orders_and_shipments.csv | Year of the shipment |
| Shipment information | Shipment Month | orders_and_shipments.csv | Month of the shipment |
| Shipment information | Shipment Day | orders_and_shipments.csv | Day of the shipment |
| Shipment information | Shipment Mode | orders_and_shipments.csv | Information on how the shipment has been dispatched, with values as First Class, Same Day, Second Class, etc. |
| Shipment information | Shipment Days - Scheduled | orders_and_shipments.csv | Information on typical amount of days needed to dispatch the goods from the moment the order has been placed |
| Warehouse | Warehouse Country | orders_and_shipments.csv | Country of the warehouse that has fulfilled this order, the only two values being Puerto Rico and USA |
| Inventory & Fulfillment | Warehouse Inventory | inventory.csv | The monthly level of inventory of a product, e.g. 930 units |
| Inventory & Fulfillment | Inventory cost per unit | inventory.csv | The monthly storage cost per unit of inventory, e.g. $2.07 |
| Inventory & Fulfillment | Warehouse Order fulfillment (days) | fulfillment.csv | The average amount of days it takes to refill stock if inventory drops below zero |


**The data can be downloaded from the sidebar on the left (under _Files_).**

## 💪 Challenge
Using either Tableau or Power BI, create an interactive dashboard to summarize your research. Things to consider:

1. Use this Workspace to prepare your data (optional).
2. Some ideas to get you started: visualize how shipments are delayed, by country, product, and over time. Analyze products by their supply versus demand ratio. Rank products by over or understock. Don't feel limited by these, you're encouraged to use your skills to consolidate as much information as possible. 
3. Create a screenshot of your (main) Tableau or Power BI dashboard, and paste that into the designated field.
4. Summarize your findings in an executive summary.

In [1]:
import pandas as pd
data = pd.read_csv("data/orders_and_shipments.csv")
data

Order ID    Order Item ID   ...   Discount %    Profit 
0           3535             8793  ...          0.25       200
1           4133            10320  ...          0.09       200
2           7396            18517  ...          0.06       200
3          11026            27608  ...          0.15       200
4          11026            27609  ...          0.13       200
...          ...              ...  ...           ...       ...
30866      73246           176561  ...          0.06         6
30867       7908            19762  ...          0.12        68
30868      29326            73368  ...          0.09        68
30869      63308           158284  ...          0.02        60
30870      67951           169874  ...           -         172

[30871 rows x 24 columns]

In [2]:
-- Query a variable:
SELECT * FROM data

Order ID    Order Item ID   ...   Discount %    Profit 
0           3535             8793  ...          0.25       200
1           4133            10320  ...          0.09       200
2           7396            18517  ...          0.06       200
3          11026            27608  ...          0.15       200
4          11026            27609  ...          0.13       200
...          ...              ...  ...           ...       ...
30866      73246           176561  ...          0.06         6
30867       7908            19762  ...          0.12        68
30868      29326            73368  ...          0.09        68
30869      63308           158284  ...          0.02        60
30870      67951           169874  ...           -         172

[30871 rows x 24 columns]

In [3]:
SELECT *
FROM data
WHERE 
    'Order ID' IS NULL
    OR 'Order Item ID' IS NULL
    OR 'Order YearMonth' IS NULL
    OR 'Order Year' IS NULL
    OR 'Order Month' IS NULL
    OR 'Order Day' IS NULL
    OR 'Order Time' IS NULL
    OR 'Order Quantity' IS NULL
    OR 'Product Department' IS NULL
    OR 'Product Category' IS NULL
    OR 'Product Name' IS NULL
    OR 'Customer ID' IS NULL
    OR 'Customer Market' IS NULL
    OR 'Customer Region' IS NULL
    OR 'Customer Country' IS NULL
    OR 'Warehouse Country' IS NULL
    OR 'Shipment Year' IS NULL
    OR 'Shipment Month' IS NULL
    OR 'Shipment Day' IS NULL
    OR 'Shipment Mode' IS NULL
    OR 'Shipment Days - Scheduled' IS NULL
    OR 'Gross Sales' IS NULL
    OR 'Discount %' IS NULL
    OR 'Profit' IS NULL;
 -- No Rows in Our Data Have missing values

Empty DataFrame
Columns: [Order ID ,  Order Item ID ,  Order YearMonth ,  Order Year ,  Order Month ,  Order Day , Order Time, Order Quantity, Product Department, Product Category, Product Name,  Customer ID , Customer Market, Customer Region, Customer Country, Warehouse Country, Shipment Year, Shipment Month, Shipment Day, Shipment Mode,  Shipment Days - Scheduled ,  Gross Sales ,  Discount % ,  Profit ]
Index: []

In [4]:
SELECT DISTINCT *
FROM data;
-- Checking for Duplicates

Order ID    Order Item ID   ...   Discount %    Profit 
0           3535             8793  ...          0.25       200
1          11026            27608  ...          0.15       200
2          11026            27609  ...          0.13       200
3          11026            27607  ...          0.16       200
4          21215            53077  ...          0.04       200
...          ...              ...  ...           ...       ...
30866      47628           119088  ...          0.01        65
30867      19096            47732  ...          0.25       150
30868       7565            18962  ...          0.07        69
30869       5991            14921  ...          0.12       125
30870      23606            59046  ...          0.05       150

[30871 rows x 24 columns]

❗️Additional Calculated Columns have been added and Data has been cleaned further in Tableau Prep

## ✍️ Judging criteria
| CATEGORY | WEIGHTING | DETAILS                                                              |
|:---------|:----------|:---------------------------------------------------------------------|
| **Visualizations** | 35% | <ul><li>Appropriateness of visualizations used.</li><li>Clarity of insight from visualizations.</li></ul> |
| **Insights** | 25%       | <ul><li>Clarity of insights - how clear and well presented the insights are.</li><li>Quality of recommendations - are appropriate analytical techniques used & are the conclusions valid?</li><li>Number of relevant insights found for the target audience.</li></ul>       |
| **Storytelling**  | 25%       | <ul><li>How well the data and insights are connected to tell a story.</li><li>How the narrative and whole report connects together.</li><li>How balanced the report is: in-depth enough but also concise.</li></ul> |
| **Votes** | 15% | <ul><li>Up voting - most upvoted entries get the most points.</li></ul> |

# **Dashboard Summary**

**KPIs** **(Late Delivery Count, Profit Margin, Gross Sales)**: The KPIs offer an at-a-glance view of important metrics. Stakeholders can monitor the number of late deliveries, assess the profitability of sales through profit margin, and track overall revenue generated through gross sales. These KPIs serve as high-level indicators of shipment performance, profitability, and business growth.

**Line Chart for Late Delivery Average Over Time**: This visualization helps stakeholders track the trend of late deliveries over time. By analyzing the patterns and identifying any spikes or dips in late delivery rates, stakeholders can uncover recurring issues and take proactive measures to improve shipment efficiency, optimize logistics, and reduce delays.

**Geo Map Showing Normalized Late Delivery by Countries**: The map visualization provides a geographical perspective on late delivery rates across different countries. Stakeholders can identify regions with higher rates of late deliveries and investigate potential causes. This information can guide decision-making related to supply chain optimization, warehouse locations, and improving logistics networks in specific areas.

**Supply/Demand Bar Graph**: This graph represents the percentage of inventory quantity compared to the order quantity. Stakeholders can identify imbalances in supply and demand for different products or categories. It can help them determine if there are overstocked items that may result in holding costs or understocked items that may lead to missed sales opportunities. By addressing these imbalances, stakeholders can optimize inventory levels, reduce costs, and improve customer satisfaction.

**Profit Forecast Line Chart**: The profit forecast line chart provides a projection of future profitability based on historical trends and patterns. Stakeholders can assess the potential impact of different factors on profitability, such as changes in sales volume, costs, or market conditions. This information enables them to make informed decisions about pricing strategies, resource allocation, and investment opportunities.

**Filters (Shipment Mode, Date, Product Department/Name)**: The inclusion of filters allows stakeholders to slice and dice the data according to their specific needs. They can explore the impact of different shipment modes, analyze trends over different time periods, and drill down into specific product departments or names. These filters enhance data exploration and facilitate deeper analysis, leading to targeted solutions for shipment and inventory management challenges.

**Horizontal Bar Chart for Top 5 Overstock and Understock Items**: This chart highlights the products that are either overstocked or understocked. Stakeholders can identify specific items that require attention and take appropriate actions to optimize inventory levels, adjust procurement strategies, or revise sales and marketing approaches. By addressing these inventory imbalances, stakeholders can minimize holding costs, improve cash flow, and enhance overall operational efficiency.

Overall, this dashboard provides comprehensive visibility into shipment and inventory management challenges, supply chain inefficiencies, and potential problems. It empowers business stakeholders to make data-driven decisions, implement structural improvements, and drive positive changes in logistics, inventory management, and overall business operations.

## 🧾 Executive summary
Data Story: Shipment and Inventory Management Insights

In our data exploration journey through Just In Time's shipment and inventory management data, we uncovered a compelling story that guided our decision-making and proposed structural business improvements. Our powerful dashboard provided a comprehensive view of key performance indicators (KPIs) and visualizations, shedding light on critical challenges and opportunities.

As we examined the data with no filters applied, we celebrated a strong profit margin above 50%, reflecting our effective cost management and revenue generation. However, the average late delivery count of 16 raised concerns about our shipment operations. This prompted us to investigate further, utilizing visualizations such as the line chart showing average late delivery over time and the geo map illustrating normalized late delivery by countries.

These visuals revealed recurring spikes in late deliveries during specific periods and highlighted regions with higher rates of late deliveries. Recognizing the need for improvement, we resolved to address the root causes of delays, optimize logistics networks, and enhance customer satisfaction.

Our dashboard's filtering capabilities became invaluable, enabling us to dive deeper into different product departments and countries. By selecting specific product departments, we gained targeted insights into inventory levels, sales performance, and profitability within each category. Additionally, analyzing performance in different countries allowed us to adapt strategies, optimize logistics, and cater to diverse market demands.

Armed with these insights, we embarked on a mission to improve our shipment and inventory management practices. We streamlined processes, strengthened partnerships, and invested in technology to enhance efficiency and meet customer expectations.

Our data-driven approach ensured that our business thrived not only in profitability but also in operational excellence. As we continue our journey, we remain committed to solving shipment and inventory challenges, analyzing supply chain inefficiencies, and proposing structural business improvements.

Through our data story, we have crafted a roadmap for success. By leveraging the power of our dashboard, we have unlocked the potential to optimize operations, drive growth, and deliver exceptional customer experiences. With data as our compass, we embrace the future with confidence, knowing that our insights will guide us toward a more efficient and effective Just In Time.



## 📷 Dashboard screenshot
![image-3](image-3.png)


## 🌐 Upload your dashboard
Tableau Public Link: https://public.tableau.com/views/SupplyChainDashboard-PremPatel/SupplyChainDashboard?:language=en-US&publish=yes&:display_count=n&:origin=viz_share_link